ambry-tools/src/main/java/com.github.ambry/clustermap/HelixBootstrapUpgradeTool.java

```diff
@@ -0,0 +1,466 @@
+   * @param zkLayoutPath the path to the zookeeper layout file.
+   * @param localDc the name of the local datacenter. This can be null.
+   * @throws IOException if there is an error reading a file.
+   * @throws JSONException if there is an error parsing the JSON content in any of the files.
+   */
+  static void bootstrapOrUpgrade(String hardwareLayoutPath, String partitionLayoutPath, String zkLayoutPath,
+      String localDc) throws IOException, JSONException {
+    HelixBootstrapUpgradeTool clusterMapToHelixMapper =
+        new HelixBootstrapUpgradeTool(hardwareLayoutPath, partitionLayoutPath, zkLayoutPath, localDc);
+    clusterMapToHelixMapper.updateHelix();
+  }
+
+  /**
+   * Instantiates this class with the given information.
+   * @param hardwareLayoutPath the path to the hardware layout file.
+   * @param partitionLayoutPath the path to the partition layout file.
+   * @param zkLayoutPath the path to the zookeeper layout file.
+   * @param localDc the name of the local datacenter. This can be null.
+   * @throws IOException if there is an error reading a file.
+   * @throws JSONException if there is an error parsing the JSON content in any of the files.
+   */
+  private HelixBootstrapUpgradeTool(String hardwareLayoutPath, String partitionLayoutPath, String zkLayoutPath,
+      String localDc) throws IOException, JSONException {
+    this.localDc = localDc;
+    parseZkJsonAndPopulateZkInfo(zkLayoutPath);
+
+    String partitionLayoutString = null;
+    try {
+      partitionLayoutString = Utils.readStringFromFile(partitionLayoutPath);
+    } catch (FileNotFoundException e) {
+      System.out.println("Partition layout path not found. Creating new file");
+    }
+    ClusterMapConfig clusterMapConfig = new ClusterMapConfig(new VerifiableProperties(new Properties()));
+    if (partitionLayoutString == null) {
+      staticClusterMap = new ClusterMapManager(new PartitionLayout(
+          new HardwareLayout(new JSONObject(Utils.readStringFromFile(hardwareLayoutPath)), clusterMapConfig)));
+    } else {
+      staticClusterMap = new ClusterMapManager(hardwareLayoutPath, partitionLayoutPath, clusterMapConfig);
+    }
+  }
+
+  /**
+   * Parses the zk layout JSON file and populates an internal map of datacenter name to Zk connect strings.
+   * @param zkLayoutPath the path to the Zookeeper layout file.
+   * @throws IOException if there is an error reading the file.
+   * @throws JSONException if there is an error parsing the JSON.
+   */
+  private void parseZkJsonAndPopulateZkInfo(String zkLayoutPath) throws IOException, JSONException {
+    JSONObject root = new JSONObject(Utils.readStringFromFile(zkLayoutPath));
+    JSONArray all = (root).getJSONArray("zkHosts");
+    for (int i = 0; i < all.length(); i++) {
+      JSONObject entry = all.getJSONObject(i);
+      dataCenterToZkAddress.put(entry.getString("datacenter"),
+          entry.getString("hostname") + ":" + entry.getString("port"));
```

Comment:
if you are going to concatenate, instead of hostname and port, why not let the user provide an endpoint?